thorlcr/graph/thgraphslave.cpp

```diff
@@ -245,6 +245,32 @@ void CSlaveActivity::setupSpace4FileStats(unsigned where, bool statsForMultipleF
         fileStats.push_back(new CRuntimeStatisticCollection(statsMapping));
 }
 
+// Collates previously collected static stats with live stats collected via a callback, and updates the activity's 'stats'
+// The callback fetches the current state of the activity's stats, called within a critical section ('statsCs'),
+// which the activity should use to protect any objects it uses whilst stats are being collected.
+void CSlaveActivity::collateStats(std::function<void(CRuntimeStatisticCollection &activeStats)> harvestStatsFn)
+{
+    if (someInactiveStats)
+    {
+        CRuntimeStatisticCollection activeStats(inactiveStats);
+        {
+            CriticalBlock block(statsCs);
+            harvestStatsFn(activeStats);
+        }
+        stats.set(activeStats);
```

Comment:
Will this cause a problem if there is another source of the stats that are merged from this source since they will be replaced?
I'm not convinced about the best solution - so discussing it below...
An alternative would be to merge into stats when inactive, and merge stats and any active into a temporary when serializing.  The issue with that is it is potentially expensive to clone information for nested scopes (e.g. timed functions).  We could introduce a serialize function which serialized merged results from two sources, but that might be more complex.

Reply:
New JIRA : https://track.hpccsystems.com/browse/HPCC-28444